src/ucp/tag/tag_recv.c

```diff
@@ -143,16 +129,27 @@ static UCS_F_ALWAYS_INLINE ucs_status_ptr_t ucp_tag_recv_common(
     req->recv.op_attr       = param->op_attr_mask;
     req->recv.tag.tag       = tag;
     req->recv.tag.tag_mask  = tag_mask;
-    if (param->op_attr_mask & UCP_OP_ATTR_FIELD_CALLBACK) {
-        req->recv.tag.cb = param->cb.recv;
-        req->user_data   = ucp_request_param_user_data(param);
-    }
 
     if (ucs_log_is_enabled(UCS_LOG_LEVEL_TRACE_REQ)) {
         req->recv.tag.info.sender_tag = 0;
     }
 
-    if (ucs_unlikely(rdesc == NULL)) {
+    if (ucs_likely(rdesc != NULL)) {
+        /* Matched */
+        if (ucs_unlikely(rdesc->flags & UCP_RECV_DESC_FLAG_RNDV)) {
+            ucp_tag_rndv_matched(worker, req,
+                                 ucp_tag_rndv_rts_from_rdesc(rdesc),
+                                 rdesc->length);
+            UCP_WORKER_STAT_RNDV(worker, RX_UNEXP, 1);
+            ucp_recv_desc_release(rdesc);
+        } else {
+            ucp_tag_recv_eager_multi(worker, req, rdesc);
+        }
+
+        if (req->flags & UCP_REQUEST_FLAG_COMPLETED) {
+            goto out_completed;
```

Comment:
i think we need to set `status = req->status` here, because we return `status` at the end of the function at line 184 (and also use it in case we need to set `param->recv.tag.info`)

Reply:
thanks, good catch!

Reply:
fixed by force push due to small pr